include/flatbuffers/util.h

```diff
@@ -335,6 +335,8 @@ inline bool StringToFloatImpl(T *val, const char *const str) {
 // - If the converted value falls out of range of corresponding return type, a
 // range error occurs. In this case value MAX(T)/MIN(T) is returned.
 template<typename T> inline bool StringToNumber(const char *s, T *val) {
+  // Assert with size_t that is a distinguish unsigned 64-bit type on Mac.
+  static_assert(sizeof(T) < sizeof(int64_t), "unexpected type T");
```

Comment:
Sorry, I don't understand what this does, and how it related to the "Mac" (Apple Clang?). This seems to prohibit T to be 64-bit, which is surprising to me.

Reply:
I'm sorry, my comment is wrong (I will change this comment.) but the problem is real.

On Mac (Darwin) `size_t` is `unsigned long` while `uint64_t` is `unsigned long long`.
Both types are 64-bit unsigned on `LP64` platform but they are not the same. So `StringToNumber<uint64_t>` doesn't catch this `size_t`.
I added this assert because it spent a lot of time to catch it with CI,
This the example: https://godbolt.org/z/Y99Pa9
This is the failure: https://github.com/google/flatbuffers/pull/6430/checks?check_run_id=1811433519

I think this `static_assert` will be sufficient. The alternative solution requires SFINAE and looks scary.

Reply:
Ok, so I guess then it should be documented that `StringToNumber` is not to be called with types like `size_t`, but only with `(u)intX_t` types?

But yes, these different ways of typedeffing 64-bit types has caused problems quite a few times before.. long vs long long is a mess. The fact that there's no foolproof way to deal with 64-bit values in C++ is crazy.

Reply:
Ah didn't see the changed comment..